Library/Homebrew/cmd/autoremove.rb

```diff
@@ -20,37 +19,9 @@ def autoremove_args
     end
   end
 
-  def get_removable_formulae(formulae)
-    removable_formulae = Formula.installed_formulae_with_no_dependents(formulae).reject do |f|
-      Tab.for_keg(f.any_installed_keg).installed_on_request
-    end
-
-    removable_formulae += get_removable_formulae(formulae - removable_formulae) if removable_formulae.present?
-
-    removable_formulae
-  end
-
   def autoremove
     args = autoremove_args.parse
 
-    removable_formulae = get_removable_formulae(Formula.installed)
-
-    if (casks = Cask::Caskroom.casks.presence)
-      removable_formulae -= casks.flat_map { |cask| cask.depends_on[:formula] }
-                                 .compact
-                                 .map { |f| Formula[f] }
-                                 .flat_map { |f| [f, *f.runtime_formula_dependencies].compact }
-    end
-    return if removable_formulae.blank?
-
-    formulae_names = removable_formulae.map(&:full_name).sort
-
-    verb = args.dry_run? ? "Would uninstall" : "Uninstalling"
-    oh1 "#{verb} #{formulae_names.count} unneeded #{"formula".pluralize(formulae_names.count)}:"
-    puts formulae_names.join("\n")
-    return if args.dry_run?
-
-    kegs_by_rack = removable_formulae.map(&:any_installed_keg).group_by(&:rack)
-    Uninstall.uninstall_kegs(kegs_by_rack)
+    Cleanup.autoremove(dry_run: args.dry_run?)
```

Comment:
Would be good to have a single autoremove integration test to hit this and the `Cleanup.autoremove` logic.

Reply:
Okay, I added two integration tests. Let me know what you think.

Reply:
Never mind I managed to squeeze them into one integration test.